VultisigApp/VultisigApp/Services/CoinService.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  CoinService.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 05/07/24.
+//
+
+import Foundation
+
+@MainActor
+class CoinService {
+    
+    static let shared = CoinService()
```

Comment:
This service doesn't have any state , it just a bunch of functions , don't make it singleton , let's create it when we need it.

Reply:
It can be static as well, but we should avoid creating a new instance every time. It interacts directly with the storage, using Swift data. Additionally, it follows the same pattern as all other services.